### DISCUSSION TITLE
Turkce Ceviri

### DISCUSSION THREAD

**Discussion by alioguzhan** [reactjs/tr.react.dev#399] [Announcements]
Soru-cevap, fikir alisverisi vs gibi konular icin. Slack kanali da var ama kac kisi hala slack kullaniyor emin degilim. Burayi kullanabiliriz gibi.

**Comment by yunusemredilber** [comment]
Türkçe çeviri için buraya yönlendirebiliriz bencede 👍🏻. Slack kanalı herkese açık olmasına gerek olmayan özel durumlar için bir seçenek olarak kalabilir belki.

**Comment by akcoren** [comment]
Merhabalar, tekil bileşenlerdeki ingilizceyi çevirecek miyiz? 
Örnek:
YouWillLearn Bileşeni sayfa içerisinde (importing-and-exporting-components.md) kullanılmasına rağmen MDX/Sandpack dizini altında tanımlanmış bağımsız bir bileşen. Çeviri kurallarında buna dair bir yönerge yok. Bu durum için nasıl bir yol izlenmeli?

**Reply by alioguzhan** [reply]
Merhaba, 

` tekil bileşenlerdeki ingilizceyi` derken tam olarak anlamadim neyi / nereyi sordugunuzu :/

**Reply by akcoren** [reply]
Bazi sayfalarin icerisindeki componentlerde de ingilizce basliklar aciklamalar mevcut. Onlari da tercume edelim mi demek istemistim ama saniyorum olusturdugum PR a gelecek olan review ile birlikte bunun da cevabini alacagim :)

**Reply by alioguzhan** [reply]
Tamam PR'i gorunce anladim :). Cevirmeniz iyi olmus onlari da. Genel fikri soyle dusunebilirsiniz:

Kod cevirilerinde, kodun calismasini bozmayacak sekilde, `stringleri` ve `commentleri` cevirebiliriz. Hatta ozel isimlerde yerellestirme bile yapabilirsiniz. Ornegin; `John` yazan bi yeri `Berkay` diye degistirebilirsiniz. Ya da `London` yazan bi yeri bizden bi sehir adi ile. Sadece; `bu degisiklik kodun calismasini bozuyor mu bozmuyor mu` buna dikkat ediyoruz. 

 Yarin review edilir muhtemelen PR

**Comment by mhasanince** [comment]
Merhaba, `See more examples below.` cumlesi bir cok yerde kullaniliyor. Bu tarz kaliplar icin ortak bir ceviri kullanalim mi? Kullanalim dersek bunu nasil yonetelim? Ortak cevirilere ekleyebiliriz bence.

**Reply by alioguzhan** [reply]
Merhaba, tabii ki olabilir. Uygun gordugunuz bir tercume varsa `TRANSLATION.md` dosyasina ekleyerek bir PR acarsaniz uzerinden konusabilirz. Bittiginde ortak cevirilere eklemis oluruz.

https://github.com/reactjs/tr.react.dev/blob/main/TRANSLATION.md

**Comment by baspinarenes** [comment]
Selamlar. https://github.com/reactjs/tr.react.dev/blob/main/TRANSLATION.md dosyasına göz attığımda bazı çevirilerin tekrar gözden geçirilmesi ve hatta çevirilmemesi gerektiğini düşünüyorum. Örneğin "yönetici", "çatı", "olay yöneticisi", "yüzeysel renderlama" ve "lazy başlatım" gibi çeviriler, okuyucunun kafasını karıştıracak ve ileri araştırma yapmayı zorlaştıracaktır. Bu gibi kritik kavramların İngilizce hallerinin kullanılması dökümanı daha anlaşılır kılmaz mı?

**Reply by baspinarenes** [reply]
Bana kalırsa eski dökümanın temel sıkıntısı buydu. Kavramları kaçırmamak için İngilizce okumak gerekiyordu.

**Reply by alioguzhan** [reply]
Selamlar. Tesekkurler yorum ve oneriler icin. Eski sitenin ilk gunlerinden beri maintainer olarak burdayim sanirim. Bahsettiginiz konuyu o zaman da slack ortaminda konusmustuk hatirladigim kadariyla. Onerdiginiz yontemin mantigini anlayabiliyorum. Hak da veriyorum sahsen. Ama burada olay `ozyinelemeli - recursive` mevzusuna geliyor yine. Soyle ki:

Bana kalirsa `recursive` de cevrilmemesi gereken bir kelime. Ama ders kitaplari dahil yaygin olarak kullaniliyor bu kelime. Ayni sekilde ornek verdiginiz kavramlardan `olay yöneticisi` ve `çatı` nin kullanildigi baya bir kaynak var sanirim. Sonuc olarak bu siteyi Turkce'ye ceviriyoruz. Her 2-3 kelimede bir ingilizce kelimelerin araya girdigi bir site cok mantikli gelmiyor bana. Bu isi yapan ve bu siteye girecek herkesin ingilizce ve proglamlama bildigini de varsayamayiz. Orijinal site de orada duruyor sonucta. Ben buradaki kaynaklarin / terimlerin yerellestirilmesinde ve cevrilmesinde cok bir sakinca gormuyorum.

Bunu soylerken; sizin bahsettiginiz durumlar / sorunlar da gayet gecerli. O yuzden bu ikisinin orta noktasini yakalamak icin en basindan beri `bazi` kavramlarin cevirisini kullanirkan parantez icinde ingilizcelerini de yaziyoruz. 

Mesela eski sitenin [Thinking in React](https://tr.legacy.reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state) sayfasinda `DRY` prensibi geciyordu. Bu isin icinde olan birisi icin gayet bilindik ve net bir kavram / kisaltma. 

Ben sadece `DRY` kavramini cevirmek icin 1 saate yakin arastirma yaptigimi hatirliyorum. En sonunda eksi sozluk'te [bir entrydeki](https://eksisozluk.com/entry/35405641) ceviri cok hosuma gitmisti `(TEK: Tekrar Etme Kendini)`. Yazarin da iznini alip oraya link vermistim siteden. Hala duruyor.

Zaten bazi kavramlari ve kelimeleri hic cevirmiyoruz. Bazilarinda da yukaridaki yontemi uyguluyoruz. Siz de uygun gordugunuz yerlerde parantez icinde ingilizcesini yazarak kafa karisikliginin onune gecebilirsiniz.

**Comment by baspinarenes** [comment]
**top level** özellikle hook'larda çok fazla karşımıza çıkan bir kavram olduğu için çevirisinin konuşulması gerektiğini düşünüyorum. Belki bu thread'de tartışabiliriz. Şahsi önerim: **kök kapsam**.

**Reply by alioguzhan** [reply]
Onceki sitede `en ust seviye` olarak cevirmistik. Direkt cevirisi zaten net oldugu icin ceviri rehberi listesine eklememisiz sanirim. Hem de hooklarin kullanimindaki limitasyonu da tanimliyor sanki. 

Bir fonksiyon/bilesen icindeki her bir loop, condition ya da baska bir fonksiyon cagrimi yeni bir alt context / seviye / branch olusturuyor. direkt fonksiyon seviyesinde olan definitionlar `en ust seviye` seklinde ifade edilebilir bence. intuition olarak da karsiliyor bu durumu.  

`kok kapsam` belki `root level` in karsiligi olabilir. yine de baskalarindan fikir ya da yorum gelirse konusalim dusunelim tabii ki.
